lms/djangoapps/grades/models.py

```diff
@@ -257,8 +257,14 @@ def save_grade(cls, **kwargs):
                     usage_key=usage_key,
                     defaults=kwargs,
                 )
+            log.info("persistent subsection grade successfully saved: {0}".format(unicode(grade)))
```

Comment:
Turns out, doing a `select_related` doesn't make this query count issue any better. Since this is a `get_or_create`, the `select_related` is attempted on the `get` (which fails), but is not included in the `create` at all. Seems to be existing django behavior: https://code.djangoproject.com/ticket/21584

Here's a sample of queries: https://gist.github.com/efischer19/3471fd87dc5b7366f23405ad87706cc7